compositionlibs/libraries/abstractcomposition/source.rb

```diff
@@ -0,0 +1,186 @@
+# frozen_string_literal: true
+
+needs 'Standard Libs/Units'
+needs 'Standard Libs/CommonInputOutputNames'
+
+needs 'CompositionLibs/ReactionComponent'
+
+needs 'Standard Libs/ItemActions'
+
+# module AbstractCompositionDefinitions
+#   include Units
+#   include CommonInputOutputNames
+
+#   ABSTRACT_COMPONENT = {
+#     input_name: ABSTRACT_NAME,
+#     qty: 5, units: MICROLITERS,
+#     sample_name: 'Optional Sample Name'
+#   }
+# end
+
+
+# Factory class for instantiating
+# @author Cannon Mallory <malloc3@uw.edu>
+# @author Devin Strickland <strklnd@uw.edu>
+
+class CompositionFactory
```

Comment:
I think maybe there should just be individual factory classes for each concrete class. I am not sure what this is accomplishing.